### PR TITLE
Edb create port

### PR DESCRIPTION
**Pull request by ring630**
new methods to create ports in EDB. Please notice, this is not duplicate methods of the ones in pyaedt/hfss3dlayout.py. The later creates ports in AEDT.

### CODE REVIEW

File: pyaedt/edb_core/hfss.py [hunk 1]
```diff
@@ -75,6 +75,39 @@ def _builder(self):
...
+        prim_id : int
+            Primitive ID.
+        point_on_edge : list
+            Coordinate of the point to define the edge terminal.
+        The point must be on the target edge but not on the two
+        ends of the edge.
+        terminal_name : str, optional
+            Name o the terminal.
+        is_ref : bool, optional
+            Where it is a reference terminal.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/hfss.py:93]:
```suggestion
            Whether it is a reference terminal.
```

File: pyaedt/edb_core/hfss.py [hunk 2]
```diff
@@ -75,6 +75,39 @@ def _builder(self):
...
+        Parameters
+        ----------
+        prim_id : int
+            Primitive ID.
+        point_on_edge : list
+            Coordinate of the point to define the edge terminal.
+        The point must be on the target edge but not on the two
+        ends of the edge.
+        terminal_name : str, optional
+            Name o the terminal.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/hfss.py:91]:
```suggestion
            Name of the terminal.
```

File: pyaedt/edb_core/hfss.py [hunk 3]
```diff
@@ -75,6 +75,39 @@ def _builder(self):
...
+    def _create_edge_terminal(self, prim_id, point_on_edge, terminal_name=None, is_ref=False):
+        """Create an edge terminal.
+
+        Parameters
+        ----------
+        prim_id : int
+            Primitive ID.
+        point_on_edge : list
+            Coordinate of the point to define the edge terminal.
+        The point must be on the target edge but not on the two
```
Comment (MaxJPRey) [inline pyaedt/edb_core/hfss.py:88]:
```suggestion
            The point must be on the target edge but not on the two
```

File: pyaedt/edb_core/hfss.py [hunk 4]
```diff
@@ -554,6 +587,92 @@ def create_edge_port_on_polygon(
...
+        """Create a vertical edge port.
+
+        Parameters
+        ----------
+        prim_id : int
+            Primitive ID.
+        point_on_edge : list
+            Coordinate of the point to define the edge terminal.
+        The point must be on the target edge but not on the two
+        ends of the edge.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/hfss.py:608]:
```suggestion
            Coordinate of the point to define the edge terminal.
            The point must be on the target edge but not on the two
            ends of the edge.
```

File: pyaedt/edb_core/hfss.py [hunk 5]
```diff
@@ -554,6 +587,92 @@ def create_edge_port_on_polygon(
...
+        """Create a horizontal edge port.
+
+        Parameters
+        ----------
+        prim_id : int
+            Primitive ID.
+        point_on_edge : list
+            Coordinate of the point to define the edge terminal.
+        The point must be on the target edge but not on the two
+        ends of the edge.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/hfss.py:649]:
```suggestion
            Coordinate of the point to define the edge terminal.
            The point must be on the target edge but not on the two
            ends of the edge.
```

File: pyaedt/edb_core/hfss.py [hunk 6]
```diff
@@ -554,6 +587,92 @@ def create_edge_port_on_polygon(
...
+        port_name : str, optional
+            Name of the port.
+        impedance : int, float, optional
+            Impedance of the port. The default value is ``50``.
+        reference_layer : str, optional
+            The reference layer of the port.
```
Comment (PipKat) [inline pyaedt/edb_core/hfss.py:614]:
```suggestion
            Reference layer of the port. The default is ``None``.
```

File: pyaedt/edb_core/hfss.py [hunk 7]
```diff
@@ -554,6 +587,92 @@ def create_edge_port_on_polygon(
...
+        Parameters
+        ----------
+        prim_id : int
+            Primitive ID.
+        point_on_edge : list
+            Coordinate of the point to define the edge terminal.
+        The point must be on the target edge but not on the two
+        ends of the edge.
+        port_name : str, optional
+            Name of the port.
```
Comment (PipKat) [inline pyaedt/edb_core/hfss.py:610]:
```suggestion
            Name of the port. The default is ``None``.
```

File: pyaedt/edb_core/hfss.py [hunk 8]
```diff
@@ -554,6 +587,92 @@ def create_edge_port_on_polygon(
...
+            Coordinate of the point to define the edge terminal.
+            The point must be on the target edge but not on the two
+            ends of the edge.
+        ref_prim_id : int
+            Reference primitive ID.
+        point_on_ref_edge : list
+            Coordinate of the point to define the reference edge
+            terminal. The point must be on the target edge but not
+            on the two ends of the edge.
+        port_name : str, optional
```
Comment (PipKat) [inline pyaedt/edb_core/hfss.py:656]:
```suggestion
        point_on_edge : list
            Coordinate of the point to define the edge terminal.
            The point must be on the target edge but not on the two
            ends of the edge.
        ref_prim_id : int, optional
            Reference primitive ID. The default is ``None``.
        point_on_ref_edge : list, optional
            Coordinate of the point to define the reference edge
            terminal. The point must be on the target edge but not
            on the two ends of the edge. The default is ``None``.
        port_name : str, optional
```

File: pyaedt/edb_core/hfss.py [hunk 9]
```diff
@@ -554,6 +587,92 @@ def create_edge_port_on_polygon(
...
+            Name of the port.
```
Comment (PipKat) [inline pyaedt/edb_core/hfss.py:657]:
```suggestion
            Name of the port. The default is ``None``.
```

**Review by MaxJPRey — APPROVED** [review]
LGTM. Thanks @ring630 for adding new features.

**Pull request merged**